"diff --git a/tensorflow/python/keras/engine/training.py b/tensorflow/python/keras/engine/training.py\nindex 32cc2a301d2..14278b25fa3 100644\n--- a/tensorflow/python/keras/engine/training.py\n+++ b/tensorflow/python/keras/engine/training.py\n@@ -87,11 +87,6 @@ try:\n   import h5py\n except ImportError:\n   h5py = None\n-\n-try:\n-  import yaml\n-except ImportError:\n-  yaml = None\n # pylint: enable=g-import-not-at-top\n \n \n@@ -2416,6 +2411,9 @@ class Model(base_layer.Layer, version_utils.ModelVersionSelector):\n   def to_yaml(self, **kwargs):\n     \"\"\"Returns a yaml string containing the network configuration.\n \n+    Note: Since TF 2.6, this method is no longer supported and will raise a\n+    RuntimeError.\n+\n     To load a network from a yaml save file, use\n     `keras.models.model_from_yaml(yaml_string, custom_objects={})`.\n \n@@ -2431,12 +2429,12 @@ class Model(base_layer.Layer, version_utils.ModelVersionSelector):\n         A YAML string.\n \n     Raises:\n-        ImportError: if yaml module is not found.\n+        RuntimeError: announces that the method poses a security risk\n     \"\"\"\n-    if yaml is None:\n-      raise ImportError(\n-          'Requires yaml module installed (`pip install pyyaml`).')\n-    return yaml.dump(self._updated_config(), **kwargs)\n+    raise RuntimeError(\n+        'Method `model.to_yaml()` has been removed due to security risk of '\n+        'arbitrary code execution. Please use `model.to_json()` instead.'\n+    )\n \n   def reset_states(self):\n     for layer in self.layers:"